src/ui/pages/SidePage/components/WalletConnect/WalletConnectStageTwo.tsx

```diff
@@ -57,13 +57,18 @@ const WalletConnectStageTwo = ({
       if (selectedIdentifier && pendingDAppMeerkat) {
         await PeerConnection.peerConnection.start(selectedIdentifier.id);
         await PeerConnection.peerConnection.connectWithDApp(pendingDAppMeerkat);
-        // Refresh the connections list
-        dispatch(
-          setWalletConnectionsCache([
-            { id: pendingDAppMeerkat, selectedAid: selectedIdentifier.id },
-            ...existingConnections,
-          ])
+        const existingConnection = existingConnections.find(
+          (connection) => connection.id === pendingDAppMeerkat
```

Comment:
Is this because the same function is used when re-connecting as when initially connecting?

Reply:
Yes, I updated this to make sure it won't create a new connection with existing meerkat